internal/paascharm/go.tmpl

```diff
@@ -0,0 +1,127 @@
+// Code generated by paascharmgen; DO NOT EDIT.
+package {{ .PackageName }}
+
+type CharmConfig struct {
+     ConfigOptions ConfigOptions
+     ProxyConfig   ProxyConfig
+     {{- if .Integrations }}
+         Integrations Integrations
+     {{- end }}
+}
+
+type ConfigOptions struct {
+     BaseURL string `env:"APP_BASE_URL"`
+     Port int `env:"APP_PORT"`
+     MetricsPort *int `env:"APP_METRICS_PORT"`
+     MetricsPath *string `env:"APP_METRICS_PATH"`
+     SecretKey *string `env:"APP_SECRET_KEY"`
+     {{- if .Options }}
+         UserConfigOptions
+     {{- end }}
+}
+
+{{ if .Options }}
+    type UserConfigOptions struct {
+    {{- range .Options }}
+        {{ .GoVarName }} {{ .GoVarType }} `env:"{{ .EnvVarName }}"`
+    {{- end }}
+    }
+{{- end }}
+
+type ProxyConfig struct {
+     HTTPProxy *string `env:"HTTP_PROXY"`
+     HTTPSProxy *string `env:"HTTPS_PROXY"`
+     NoProxy []string `env:"NO_PROXY"`
+}
+
+{{ if .Integrations }}
+   type Integrations struct {
+   {{- range .Integrations }}
+       {{ .GoName }} {{ .GoName }}Integration {{ if .IsDatabase }}`envPrefix:"{{ .DatabasePrefix }}"`{{ end }}
+   {{- end }}
+   }
+{{ end }}
+
+
+{{ if .HasDatabaseIntegrations }}
+
+   {{- range .Integrations }}
+       {{ if .IsDatabase }}
+       type {{ .GoName }}Integration struct {
+            DatabaseIntegration
+       }
+       {{ end }}
+    {{ end }}
+
+    type DatabaseIntegration struct {
```

Comment:
We should probably specify that the struct here is always used with a prefix in the generated code, in the case there are multiple databases